with_locked_file

rixed opened the pull request:
Proposed changes on top of https://github.com/ocaml-batteries-team/batteries-included/pull/512

UnixJunkie commented:
Could there be a test?
There is a typo in the ocamldoc: "whole the file" -> "the whole file".
Can you update the changelog?

rixed commented:
Rebased, fixed typo, modified changelog.
As said in the commit message, I have no idea how to test this meaningfully in a unit test, as a single process cannot exclude itself with lockf.
